src/test/signalk/SKNMEA2000ParserTest.cpp

```diff
@@ -83,5 +83,78 @@ TEST_CASE("SKNMEA2000Parser: Basic tests") {
+    SetN2kMagneticHeading(msg, 0, 0.7898, -1.5883, 0.014); // 181°, -2°, 3.2°
+    const SKUpdate &update = p.parse(SKSourceInputNMEA2000, msg, SKTime(0));
+    CHECK( update.getSize() == 2);
+    CHECK( (update.getNavigationHeadingMagnetic() - 0.7898) < 0.01 );
+    // CHECK( update.getNavigationMagneticDeviation() == -1.5883 );
+    CHECK( update.getNavigationMagneticVariation() == 0.014 );
+  }
+
+  SECTION("127245: Rudder Angle") {
+    SetN2kRudder(msg, SKDegToRad(-3.6), 0, N2kRDO_NoDirectionOrder, N2kDoubleNA);
+    const SKUpdate &update = p.parse(SKSourceInputNMEA2000, msg, SKTime(0));
+    CHECK( update.getSize() == 1);
+    // Some precision is lost because NMEA2000 rounds this value to 0.0001 precision
+    CHECK( update.getSteeringRudderAngle() == Approx(SKDegToRad(-3.6)).epsilon(0.0001) );
+  }
+
+  SECTION("130306: Ground Wind Test TWS, TWD TrueNorth referenced") {
+    SetN2kWindSpeed(msg, 0, 16, SKDegToRad(45), N2kWind_True_North);
+    const SKUpdate &update = p.parse(SKSourceInputNMEA2000, msg, SKTime(0));
+    CHECK( update.getSize() == 2);
+    CHECK( update.getEnvironmentWindSpeedOverGround() == 16 );
+    CHECK( update.getEnvironmentWindDirectionTrue() == Approx(SKDegToRad(45)) );
+  }
+
+  SECTION("130306: Ground Wind Test TWS, TWD magnetic North referenced") {
+    SetN2kWindSpeed(msg, 0, 16, SKDegToRad(45), N2kWind_Magnetic);
+    const SKUpdate &update = p.parse(SKSourceInputNMEA2000, msg, SKTime(0));
+    CHECK( update.getSize() == 2);
+    CHECK( update.getEnvironmentWindSpeedOverGround() == 16 );
+    CHECK( update.getEnvironmentWindDirectionMagnetic() == Approx(SKDegToRad(45)) );
+  }
+
+  SECTION("130306: Apparent Wind AWS, AWA Test starboard") {
+    //TODO: watch if Timo will correct Typo
+    SetN2kWindSpeed(msg, 0, 12.4, SKDegToRad(29.8), N2kWind_Apprent);
+    const SKUpdate &update = p.parse(SKSourceInputNMEA2000, msg, SKTime(0));
+    CHECK( update.getSize() == 2);
+    CHECK( update.getEnvironmentWindSpeedApparent() == 12.4 );
+    CHECK( update.getEnvironmentWindAngleApparent() == Approx(SKDegToRad(29.8)) );
+  }
+
+  SECTION("130306: Apparent Wind AWS, AWA Test port") {
+    //TODO: watch if Timo will correct Typo
+    SetN2kWindSpeed(msg, 0, 12.4, SKDegToRad(330), N2kWind_Apprent);
+    const SKUpdate &update = p.parse(SKSourceInputNMEA2000, msg, SKTime(0));
+    CHECK( update.getSize() == 2);
+    CHECK( update.getEnvironmentWindSpeedApparent() == 12.4 );
+    CHECK( update.getEnvironmentWindAngleApparent() == Approx(SKDegToRad(-30)) );
+  }
+
+  SECTION("130306: Ground Wind Speed and Angle TWS, TWA port") {
+    SetN2kWindSpeed(msg, 0, 12.4, SKDegToRad(359), N2kWind_True_boat);
+    const SKUpdate &update = p.parse(SKSourceInputNMEA2000, msg, SKTime(0));
+    CHECK( update.getSize() == 2);
+    CHECK( update.getEnvironmentWindSpeedOverGround() == 12.4 );
+    CHECK( update.getEnvironmentWindAngleTrueGround() == Approx(SKDegToRad(-1)).epsilon(0.0001) );
```

Comment:
Added epsilon here because this test did not pass.

We lose some precision when we convert data to NMEA2000 (because NMEA2000 uses less bytes for double - and because the library automatically rounds values to 0.0001 for 2bytesdouble).